src/tokenizer.js

```diff
@@ -769,6 +745,51 @@ export default class Tokenizer {
     return r1 << 4 | r2;
   }
 
+  scanUnicode() {
```

Comment:
This method looks a bit too long for its purpose. Can you abstract it a little bit (say turn the second one into a loop, then the two branches can be merged)?